daemon/api.go

```diff
@@ -631,24 +638,58 @@ func (inst *snapInstruction) Agreed(intro, license string) bool {
 	return true
 }
 
-var snappyInstall = snappy.Install
+var snapstateInstall = snapstate.Install
+
+func waitChange(chg *state.Change) error {
+	for {
+		chg.State().Lock()
+		status := chg.Status()
+		err := chg.Err()
+		chg.State().Unlock()
```

Comment:
This is a bit boring to compute unnecessarily inside a busy loop. I suggest something like this below:
```
st := chg.State()
for {
        st.Lock()
        s := chg.Status()
        if s == DoneStatus || s == ErrorStatus {
                defer st.Unlock()
                return chg.Err()
        }
        st.Unlock()
        time.Sleep(250 * time.Millisecond)
}
```

I'll cook a proper waiter that is not a busy loop soon.